consensus/event_buffer.go

```diff
@@ -0,0 +1,45 @@
+package consensus
+
+import (
+	"github.com/tendermint/tendermint/types"
+)
+
+const (
+	eventBufferCapacity = 1000
+)
+
+// Interface assertions
+var _ types.EventsPublisher = (*EventBuffer)(nil)
+
+// EventBuffer is a buffer of events.
+type EventBuffer struct {
```

Comment:
I'd put the buffering into the pubsub package as well, as it is a standard utility all clients would want.